fastlane/lib/fastlane/actions/upload_app_data_usage_to_app_store.rb

```diff
@@ -0,0 +1,316 @@
+        UI.message("Login to App Store Connect (#{params[:username]})")
+        Spaceship::ConnectAPI.login(params[:username], use_portal: false, use_tunes: true, tunes_team_id: params[:team_id], team_name: params[:team_name])
+        UI.message("Login successful")
+
+        # Get App
+        app = Spaceship::ConnectAPI::App.find(params[:app_identifier])
+        unless app
+          UI.user_error!("Could not find app with bundle identifier '#{params[:app_identifier]}' on account #{params[:username]}")
+        end
+
+        # Attempt to load JSON file
+        usages_config = load_json_file(params)
+
+        # Start interactive questions to generate and save JSON file
+        unless usages_config
+          usages_config = ask_interactive_questions_for_json
+
+          json = JSON.pretty_generate(usages_config)
+          path = output_path(params)
+
+          UI.message("Writing file to #{path}")
+          File.write(path, json)
+        end
+
+        # Process JSON file to save app data usages to API
+        upload_app_data_usages(params, app, usages_config)
+      end
+
+      def self.load_json_file(params)
+        path = params[:json_path]
+        return nil if path.nil?
+        return nil unless File.exist?(path)
+        return JSON.parse(File.read(path))
+      end
+
+      def self.output_path(params)
+        path = params[:output_json_path] || DEFAULT_PATH
+        path = File.absolute_path(path)
+        return File.join(path, DEFAULT_FILE_NAME)
+      end
+
+      def self.ask_interactive_questions_for_json
+        UI.important("You did not provide a JSON file for updating the app data usages")
+        UI.important("fastlane will now run you through interactive question to generate the JSON file")
+        UI.important("")
+        UI.important("This JSON file can be saved in source control and used in this action with the :json_file option")
+
+        unless UI.confirm("Ready to start?")
+          UI.user_error!("Cancelled")
+        end
+
+        # Fetch categories and purposes used for generating interactive questions
+        categories = Spaceship::ConnectAPI::AppDataUsageCategory.all(includes: "grouping")
+        purposes = Spaceship::ConnectAPI::AppDataUsagePurpose.all
+
+        json = []
+
+        unless UI.confirm("Are you collecting data?")
+          json << {
+            "data_protections" => ["DATA_NOT_COLLECTED"]
+          }
+
+          return json
+        end
+
+        categories.each do |category|
+          # Ask if using category
+          next unless UI.confirm("Collect data for #{category.id}?")
+
+          purpose_names = purposes.map(&:id).join(', ')
+          UI.message("How will this data be used? You'll be offered with #{purpose_names}")
+
+          # Ask purposes
+          selected_purposes = []
+          purposes.each do |purpose|
+            selected_purposes << purpose if UI.confirm("Used for #{purpose.id}?")
+          end
```

Comment:
What happens if I select no for all of them? It sounds like based on some comments later that that would mean that category is not used (i.e. yes to "collect data?" + no to all "used for purpose?" == no to "collect data?"), so perhaps it would be good to add an additional check afterwards if `selected_purposes` is empty to ensure that's what the user wanted